### PR TITLE
Lawnmowers can cut long grass

### PR DESCRIPTION
**Pull request by ghost**
#### Summary
```SUMMARY: Features "Lawnmowers can cut long grass"```

#### Purpose of change
Every single lawnmower found in the game was broken. this was a bit weird.
Also cutting long grass is laborious and requires weird tools, and long grass is annoying.
This enables lawnmowers to cut grass

#### Describe the solution
Made the lawnmower item that spawns in the world into a folding vehicle item.
Unfold it to get a little engine-powered lawnmower, turn it on, turn the blades on, drag the vehicle  to cut long grass.
There is the actual part : lawnmower blade, which can be attached to any other vehicle, and works when the vehicle is being driven and does the same thing.

#### Describe alternatives you've considered
I looked at the other things like PLOW which could possibly be made to work whilst vehicle is being dragged, but thats for another PR.

#### Additional context
For now, this just adds a little bit to the strength requirement to drag the vehicle to simulate the added resistance, and it takes a little bit of powerr off of an engine.
In the future if dragged plows are added, itll need to add some more indepth simulation to allow it to work properly with the calorie usage changes coming.

### CODE REVIEW

File: data/json/items/generic.json [hunk 1]
```diff
@@ -688,20 +688,26 @@
...
     "material": [ "steel", "iron" ],
     "weight": 22680,
     "volume": 25,
     "bashing": 6,
-    "to_hit": -4
+    "to_hit": -4,
+    "use_action": {
+      "type": "unfold_vehicle",
+      "vehicle_name": "lawnmower",
+      "unfold_msg": "You set up the lawnmower and unfold it's handle.",
```
Comment (Night-Pryanik) [inline data/json/items/generic.json]:
```suggestion
      "unfold_msg": "You set up the lawnmower and unfold its handle.",
```

**Comment by ifreund** [comment]
I'm not actually sure that the typical in-game lawnmower can cut long grass. Typical lawnmowers don't really work for grass longer than about 10 inches or so, as the strain on the engine becomes too high (10 inches is probably a generous estimate).  To mow tall grass you really need something like a [field trimmer](https://www.amazon.com/s?k=field+trimmer) or [brush mower](https://www.drpower.com/power-equipment/field-brush-mowers/).  As long grass is described as `"Long shaggy grass about shin high."` and tall grass as `"Tall grass about head high."` I'm a little skeptical if our current lawnmowers would work well, or at all.

**Comment by EirenexTheDragon** [comment]
how about something larger, like this one? 
grafik

**Comment by davidpwbrown** [comment]
Why is tall grass about head high??? is that a thing in New England? or a cata-thing?

**Comment by ifreund** [comment]
>Why is tall grass about head high??? is that a thing in New England? or a cata-thing?

I have no idea 😄

**Comment by JonahDes** [comment]
I've personally used hedge trimmers in paid work clearing people's gardens. You take it out by the root, for anything that's like really overgrown, over a metre, usually. Then people use a mower to cut that into shape. I suppose you can do it the other way, it's not like the lawnmower is going to care about it. You would have to re-empty it often.

**Comment by kevingranade** [comment]
We can revisit the definition and placement of long grass, but as defined now, it's not something that an ordinary mower can handle.

**Comment by davidpwbrown** [comment]
Are we attached to tall-grass being depicted as "head-high"? 
Would it be preferable if it's description was changed to hip-high or something, as long-grass is described as shin-high.

Would then t_grass_long and t_grass_tall both be mowable?

**Comment by davidpwbrown** [comment]
To be clear t_grass_long is the one described as shin high, and shin high is fine for mowing? and t_grass_tall is the problem as its currently described as "head-high"

**Comment by davidpwbrown** [comment]
Ok yeah, tall grass is way too tall in-game, but ifreund has shown me research showing that your average mower struggles with even knee-high grass without other techniques and tools to assist.
I did not expect to spend my Sunday studying mowing techniques, but there it is. 
I could probably come back to this and add some kind of ultra-death-mower that handles tall grass properly, but as things stand, the standard mower just aint good enough.

**Pull request closed without merging**